Scripts/RemediationScripts/Remediate-ConfigureAADAuthExtOnLinuxVMSS.ps1

```diff
@@ -0,0 +1,1109 @@
+
+        $NonCompliantVMSSDetails | Export-CSV -Path $backupFile -NoTypeInformation
+
+        Write-Host "Virtual machines Scale Set(s) details have been backed up to" -NoNewline
+        Write-Host " [$($backupFile)]" -ForegroundColor $([Constants]::MessageType.Update)
+    }
+    else
+    {
+        Write-Host "Skipped as -FilePath is provided" -ForegroundColor $([Constants]::MessageType.Warning)
+    }
+
+    if (-not $DryRun)
+    {
+        Write-Host $([Constants]::DoubleDashLine)
+        Write-Host "[Step 4 of 4] Remediating non compliant Linux Virtual Machine Scale Sets..." 
+        Write-Host $([Constants]::SingleDashLine)
+        
+        if (-not $Force)
+        {
+            Write-Host "Found total [$($NonCompliantVMSSDetails.count)] Virtual machines Scale Set(s) where AAD Extension is not present." -ForegroundColor $([Constants]::MessageType.Warning)
+            Write-Host "This step will add AAD extension for all non-complaint Virtual Machine Scale Set(s)." -ForegroundColor $([Constants]::MessageType.Warning)
+            Write-Host "Do you want to Continue? " -ForegroundColor $([Constants]::MessageType.Warning)
+            
+            $userInput = Read-Host -Prompt "(Y|N)"
+
+            if($userInput -ne "Y")
+            {
+                Write-Host "AAD Extension will not be added  to the Virtual machines Scale Set(s) in the Subscription. Exiting..." -ForegroundColor $([Constants]::MessageType.Warning)
+                break
+            }
+        }
+        else
+        {
+            Write-Host "'Force' flag is provided. AAD extension will be added for Virtual Machine Scale Set(s) in the Subscription without any further prompts." -ForegroundColor $([Constants]::MessageType.Warning)
+        }
+
+        # List for storing remediated Virtual Machines Scale set(s)
+        $VMSSRemediated = @()
+
+        # List for storing skipped Virtual Machines Scale Set(s)
+        $VMSSSkipped = @()
+
+        Write-Host "Adding AAD Extension on all non compliant Virtual Machines Scale Set(s)." -ForegroundColor $([Constants]::MessageType.Info)
+        Write-Host $([Constants]::SingleDashLine)
+
+        # Loop through the list of Virtual Machines Scale Set(s) which needs to be remediated.
+        $NonCompliantVMSSDetails | ForEach-Object{
+            $VMSS = $_
+            $VMSS | Add-Member -NotePropertyName isExtInstalledPostRemediation -NotePropertyValue $false
+
+            Write-Host "Adding AAD Auth Extension on [$($_.ResourceName)]." -ForegroundColor $([Constants]::MessageType.Info)
+            try
+            {
+                    # Remediation command starts from here
+                    $vmssInstance = Get-AzVmss -ResourceGroupName $VMSS.ResourceGroupName -VMScaleSetName $VMSS.ResourceName
+
+                    Add-AzVmssExtension -VirtualMachineScaleSet $vmssInstance -Name $reqExtensionName -Publisher $reqExtPublisher -Type $reqExtensionType -TypeHandlerVersion 1.0
```

Comment:
Is -TypeHandlerVersion 1.0 mandatory, if yes, then is 1.0 is latets version?

Reply:
TypeHandlerVersion is mandatory, also there is only one version available that is 1.0